Compilig with ESPHOME 2022.05.0 fails

Compiling fails with latest ESPHOME version.

Three compiling logs below

- 2022.04.0, success
- 2022.05.0, failed
- 2022.05.0 with clean first, failed

```
hp@hpix:~$ pip install esphome==2022.04.0
Defaulting to user installation because normal site-packages is not writeable
Collecting esphome==2022.04.0
  Using cached esphome-2022.4.0-py2.py3-none-any.whl (2.2 MB)
Requirement already satisfied: esptool==3.3 in ./.local/lib/python3.10/site-packages (from esphome==2022.04.0) (3.3)
Requirement already satisfied: pyserial==3.5 in /usr/lib/python3/dist-packages (from esphome==2022.04.0) (3.5)
Collecting click==8.1.2
  Using cached click-8.1.2-py3-none-any.whl (96 kB)
Requirement already satisfied: tzlocal==4.2 in ./.local/lib/python3.10/site-packages (from esphome==2022.04.0) (4.2)
Requirement already satisfied: paho-mqtt==1.6.1 in ./.local/lib/python3.10/site-packages (from esphome==2022.04.0) (1.6.1)
Requirement already satisfied: aioesphomeapi==10.8.2 in ./.local/lib/python3.10/site-packages (from esphome==2022.04.0) (10.8.2)
Requirement already satisfied: zeroconf==0.38.4 in ./.local/lib/python3.10/site-packages (from esphome==2022.04.0) (0.38.4)
Requirement already satisfied: kconfiglib==13.7.1 in ./.local/lib/python3.10/site-packages (from esphome==2022.04.0) (13.7.1)
Requirement already satisfied: colorama==0.4.4 in /usr/lib/python3/dist-packages (from esphome==2022.04.0) (0.4.4)
Requirement already satisfied: voluptuous==0.13.1 in ./.local/lib/python3.10/site-packages (from esphome==2022.04.0) (0.13.1)
Requirement already satisfied: tzdata>=2021.1 in ./.local/lib/python3.10/site-packages (from esphome==2022.04.0) (2022.1)
Requirement already satisfied: tornado==6.1 in ./.local/lib/python3.10/site-packages (from esphome==2022.04.0) (6.1)
Requirement already satisfied: platformio==5.2.5 in ./.local/lib/python3.10/site-packages (from esphome==2022.04.0) (5.2.5)
Collecting esphome-dashboard==20220309.0
  Using cached esphome_dashboard-20220309.0-py3-none-any.whl (593 kB)
Requirement already satisfied: PyYAML==6.0 in ./.local/lib/python3.10/site-packages (from esphome==2022.04.0) (6.0)
Requirement already satisfied: protobuf<4.0,>=3.12.2 in ./.local/lib/python3.10/site-packages (from aioesphomeapi==10.8.2->esphome==2022.04.0) (3.20.1)
Requirement already satisfied: noiseprotocol<1.0,>=0.3.1 in ./.local/lib/python3.10/site-packages (from aioesphomeapi==10.8.2->esphome==2022.04.0) (0.3.1)
Requirement already satisfied: cryptography>=2.1.4 in /usr/lib/python3/dist-packages (from esptool==3.3->esphome==2022.04.0) (3.4.8)
Requirement already satisfied: bitstring>=3.1.6 in ./.local/lib/python3.10/site-packages (from esptool==3.3->esphome==2022.04.0) (3.1.9)
Requirement already satisfied: reedsolo<=1.5.4,>=1.5.3 in ./.local/lib/python3.10/site-packages (from esptool==3.3->esphome==2022.04.0) (1.5.4)
Requirement already satisfied: ecdsa>=0.16.0 in ./.local/lib/python3.10/site-packages (from esptool==3.3->esphome==2022.04.0) (0.17.0)
Requirement already satisfied: requests==2.* in /usr/lib/python3/dist-packages (from platformio==5.2.5->esphome==2022.04.0) (2.25.1)
Requirement already satisfied: semantic-version==2.9.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.04.0) (2.9.0)
Requirement already satisfied: pyelftools<1,>=0.27 in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.04.0) (0.28)
Requirement already satisfied: uvicorn==0.17.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.04.0) (0.17.6)
Requirement already satisfied: ajsonrpc==1.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.04.0) (1.2.0)
Requirement already satisfied: marshmallow<4,>=2 in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.04.0) (3.15.0)
Requirement already satisfied: tabulate==0.8.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.04.0) (0.8.9)
Requirement already satisfied: wsproto==1.0.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.04.0) (1.0.0)
Requirement already satisfied: starlette==0.18.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.04.0) (0.18.0)
Requirement already satisfied: bottle==0.12.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.04.0) (0.12.19)
Requirement already satisfied: aiofiles==0.8.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.04.0) (0.8.0)
Requirement already satisfied: pytz-deprecation-shim in ./.local/lib/python3.10/site-packages (from tzlocal==4.2->esphome==2022.04.0) (0.1.0.post0)
Requirement already satisfied: ifaddr>=0.1.7 in ./.local/lib/python3.10/site-packages (from zeroconf==0.38.4->esphome==2022.04.0) (0.1.7)
Requirement already satisfied: anyio<4,>=3.0.0 in ./.local/lib/python3.10/site-packages (from starlette==0.18.*->platformio==5.2.5->esphome==2022.04.0) (3.6.1)
Requirement already satisfied: h11>=0.8 in ./.local/lib/python3.10/site-packages (from uvicorn==0.17.*->platformio==5.2.5->esphome==2022.04.0) (0.13.0)
Requirement already satisfied: asgiref>=3.4.0 in ./.local/lib/python3.10/site-packages (from uvicorn==0.17.*->platformio==5.2.5->esphome==2022.04.0) (3.5.2)
Requirement already satisfied: six>=1.9.0 in /usr/lib/python3/dist-packages (from ecdsa>=0.16.0->esptool==3.3->esphome==2022.04.0) (1.16.0)
Requirement already satisfied: packaging in ./.local/lib/python3.10/site-packages (from marshmallow<4,>=2->platformio==5.2.5->esphome==2022.04.0) (21.3)
Requirement already satisfied: idna>=2.8 in /usr/lib/python3/dist-packages (from anyio<4,>=3.0.0->starlette==0.18.*->platformio==5.2.5->esphome==2022.04.0) (3.3)
Requirement already satisfied: sniffio>=1.1 in ./.local/lib/python3.10/site-packages (from anyio<4,>=3.0.0->starlette==0.18.*->platformio==5.2.5->esphome==2022.04.0) (1.2.0)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/lib/python3/dist-packages (from packaging->marshmallow<4,>=2->platformio==5.2.5->esphome==2022.04.0) (2.4.7)
Installing collected packages: esphome-dashboard, click, esphome
  Attempting uninstall: esphome-dashboard
    Found existing installation: esphome-dashboard 20220508.0
    Uninstalling esphome-dashboard-20220508.0:
      Successfully uninstalled esphome-dashboard-20220508.0
  Attempting uninstall: click
    Found existing installation: click 8.1.3
    Uninstalling click-8.1.3:
      Successfully uninstalled click-8.1.3
  Attempting uninstall: esphome
    Found existing installation: esphome 2022.5.0
    Uninstalling esphome-2022.5.0:
      Successfully uninstalled esphome-2022.5.0
Successfully installed click-8.1.2 esphome-2022.4.0 esphome-dashboard-20220309.0
hp@hpix:~$ esphome compile livingroom-ac.yaml
INFO Reading configuration livingroom-ac.yaml...
INFO Detected timezone 'Etc/UTC'
INFO Generating C++ source...
INFO Compiling app...
Processing living-room-ac (board: esp32cam; framework: arduino; platform: platformio/espressif32 @ 3.5.0)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
Dependency Graph
|-- <WiFi> 1.0
|-- <ESPmDNS> 1.0
|-- <Update> 1.0
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_connection.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_frame_helper.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_pb2.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_pb2_service.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_server.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/list_entities.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/proto.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/subscribe_state.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/user_services.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/binary_sensor/automation.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/binary_sensor/binary_sensor.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/binary_sensor/filter.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/climate/climate.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/esp32/preferences.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/integration/integration_sensor.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/logger/logger.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_component.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_esp32_arduino.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_esp8266.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_esp_idf.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/network/util.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_backend_arduino_esp32.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_backend_arduino_esp8266.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_backend_esp_idf.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_component.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/panasonic_ac/esppac.cpp.o
RAM:   [==        ]  16.7% (used 54708 bytes from 327680 bytes)
Flash: [========= ]  88.0% (used 1614774 bytes from 1835008 bytes)
===================================================================== [SUCCESS] Took 8.76 seconds =====================================================================
INFO Successfully compiled program.
hp@hpix:~$ pip install esphome==2022.05.0
Defaulting to user installation because normal site-packages is not writeable
Collecting esphome==2022.05.0
  Using cached esphome-2022.5.0-py2.py3-none-any.whl (2.3 MB)
Requirement already satisfied: tzlocal==4.2 in ./.local/lib/python3.10/site-packages (from esphome==2022.05.0) (4.2)
Requirement already satisfied: platformio==5.2.5 in ./.local/lib/python3.10/site-packages (from esphome==2022.05.0) (5.2.5)
Requirement already satisfied: pyserial==3.5 in /usr/lib/python3/dist-packages (from esphome==2022.05.0) (3.5)
Collecting esphome-dashboard==20220508.0
  Using cached esphome_dashboard-20220508.0-py3-none-any.whl (601 kB)
Requirement already satisfied: colorama==0.4.4 in /usr/lib/python3/dist-packages (from esphome==2022.05.0) (0.4.4)
Requirement already satisfied: paho-mqtt==1.6.1 in ./.local/lib/python3.10/site-packages (from esphome==2022.05.0) (1.6.1)
Requirement already satisfied: tornado==6.1 in ./.local/lib/python3.10/site-packages (from esphome==2022.05.0) (6.1)
Requirement already satisfied: aioesphomeapi==10.8.2 in ./.local/lib/python3.10/site-packages (from esphome==2022.05.0) (10.8.2)
Requirement already satisfied: esptool==3.3 in ./.local/lib/python3.10/site-packages (from esphome==2022.05.0) (3.3)
Collecting click==8.1.3
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Requirement already satisfied: kconfiglib==13.7.1 in ./.local/lib/python3.10/site-packages (from esphome==2022.05.0) (13.7.1)
Requirement already satisfied: tzdata>=2021.1 in ./.local/lib/python3.10/site-packages (from esphome==2022.05.0) (2022.1)
Requirement already satisfied: voluptuous==0.13.1 in ./.local/lib/python3.10/site-packages (from esphome==2022.05.0) (0.13.1)
Requirement already satisfied: zeroconf==0.38.4 in ./.local/lib/python3.10/site-packages (from esphome==2022.05.0) (0.38.4)
Requirement already satisfied: PyYAML==6.0 in ./.local/lib/python3.10/site-packages (from esphome==2022.05.0) (6.0)
Requirement already satisfied: protobuf<4.0,>=3.12.2 in ./.local/lib/python3.10/site-packages (from aioesphomeapi==10.8.2->esphome==2022.05.0) (3.20.1)
Requirement already satisfied: noiseprotocol<1.0,>=0.3.1 in ./.local/lib/python3.10/site-packages (from aioesphomeapi==10.8.2->esphome==2022.05.0) (0.3.1)
Requirement already satisfied: cryptography>=2.1.4 in /usr/lib/python3/dist-packages (from esptool==3.3->esphome==2022.05.0) (3.4.8)
Requirement already satisfied: ecdsa>=0.16.0 in ./.local/lib/python3.10/site-packages (from esptool==3.3->esphome==2022.05.0) (0.17.0)
Requirement already satisfied: bitstring>=3.1.6 in ./.local/lib/python3.10/site-packages (from esptool==3.3->esphome==2022.05.0) (3.1.9)
Requirement already satisfied: reedsolo<=1.5.4,>=1.5.3 in ./.local/lib/python3.10/site-packages (from esptool==3.3->esphome==2022.05.0) (1.5.4)
Requirement already satisfied: bottle==0.12.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.05.0) (0.12.19)
Requirement already satisfied: starlette==0.18.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.05.0) (0.18.0)
Requirement already satisfied: requests==2.* in /usr/lib/python3/dist-packages (from platformio==5.2.5->esphome==2022.05.0) (2.25.1)
Requirement already satisfied: marshmallow<4,>=2 in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.05.0) (3.15.0)
Requirement already satisfied: pyelftools<1,>=0.27 in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.05.0) (0.28)
Requirement already satisfied: tabulate==0.8.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.05.0) (0.8.9)
Requirement already satisfied: wsproto==1.0.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.05.0) (1.0.0)
Requirement already satisfied: uvicorn==0.17.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.05.0) (0.17.6)
Requirement already satisfied: aiofiles==0.8.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.05.0) (0.8.0)
Requirement already satisfied: ajsonrpc==1.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.05.0) (1.2.0)
Requirement already satisfied: semantic-version==2.9.* in ./.local/lib/python3.10/site-packages (from platformio==5.2.5->esphome==2022.05.0) (2.9.0)
Requirement already satisfied: pytz-deprecation-shim in ./.local/lib/python3.10/site-packages (from tzlocal==4.2->esphome==2022.05.0) (0.1.0.post0)
Requirement already satisfied: ifaddr>=0.1.7 in ./.local/lib/python3.10/site-packages (from zeroconf==0.38.4->esphome==2022.05.0) (0.1.7)
Requirement already satisfied: anyio<4,>=3.0.0 in ./.local/lib/python3.10/site-packages (from starlette==0.18.*->platformio==5.2.5->esphome==2022.05.0) (3.6.1)
Requirement already satisfied: h11>=0.8 in ./.local/lib/python3.10/site-packages (from uvicorn==0.17.*->platformio==5.2.5->esphome==2022.05.0) (0.13.0)
Requirement already satisfied: asgiref>=3.4.0 in ./.local/lib/python3.10/site-packages (from uvicorn==0.17.*->platformio==5.2.5->esphome==2022.05.0) (3.5.2)
Requirement already satisfied: six>=1.9.0 in /usr/lib/python3/dist-packages (from ecdsa>=0.16.0->esptool==3.3->esphome==2022.05.0) (1.16.0)
Requirement already satisfied: packaging in ./.local/lib/python3.10/site-packages (from marshmallow<4,>=2->platformio==5.2.5->esphome==2022.05.0) (21.3)
Requirement already satisfied: idna>=2.8 in /usr/lib/python3/dist-packages (from anyio<4,>=3.0.0->starlette==0.18.*->platformio==5.2.5->esphome==2022.05.0) (3.3)
Requirement already satisfied: sniffio>=1.1 in ./.local/lib/python3.10/site-packages (from anyio<4,>=3.0.0->starlette==0.18.*->platformio==5.2.5->esphome==2022.05.0) (1.2.0)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/lib/python3/dist-packages (from packaging->marshmallow<4,>=2->platformio==5.2.5->esphome==2022.05.0) (2.4.7)
Installing collected packages: esphome-dashboard, click, esphome
  Attempting uninstall: esphome-dashboard
    Found existing installation: esphome-dashboard 20220309.0
    Uninstalling esphome-dashboard-20220309.0:
      Successfully uninstalled esphome-dashboard-20220309.0
  Attempting uninstall: click
    Found existing installation: click 8.1.2
    Uninstalling click-8.1.2:
      Successfully uninstalled click-8.1.2
  Attempting uninstall: esphome
    Found existing installation: esphome 2022.4.0
    Uninstalling esphome-2022.4.0:
      Successfully uninstalled esphome-2022.4.0
Successfully installed click-8.1.3 esphome-2022.5.0 esphome-dashboard-20220508.0
hp@hpix:~$ esphome compile livingroom-ac.yaml
INFO Reading configuration livingroom-ac.yaml...
INFO Detected timezone 'Etc/UTC'
INFO Generating C++ source...
INFO Compiling app...
Processing living-room-ac (board: esp32cam; framework: arduino; platform: platformio/espressif32 @ 3.5.0)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
Dependency Graph
|-- <WiFi> 1.0
|-- <ESPmDNS> 1.0
|-- <Update> 1.0
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_connection.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_frame_helper.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_pb2.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_pb2_service.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_server.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/list_entities.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/proto.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/subscribe_state.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/user_services.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/binary_sensor/automation.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/binary_sensor/binary_sensor.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/binary_sensor/filter.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/climate/climate.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/esp32/preferences.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/integration/integration_sensor.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/logger/logger.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_component.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_esp32_arduino.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_esp8266.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_esp_idf.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/network/util.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_backend_arduino_esp32.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_backend_arduino_esp8266.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_backend_esp_idf.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_component.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/panasonic_ac/esppac.cpp.o
src/esphome/components/panasonic_ac/esppac.cpp: In member function 'void esphome::panasonic_ac::PanasonicAC::set_vertical_swing_select(esphome::select::Select*)':
src/esphome/components/panasonic_ac/esppac.cpp:187:4: error: no matching function for call to 'esphome::select::Select::add_on_state_callback(esphome::panasonic_ac::PanasonicAC::set_vertical_swing_select(esphome::select::Select*)::<lambda(const string&)>)'
   });
    ^
In file included from src/esphome/components/panasonic_ac/esppac.h:4:0,
                 from src/esphome/components/panasonic_ac/esppac.cpp:1:
src/esphome/components/select/select.h:55:8: note: candidate: void esphome::select::Select::add_on_state_callback(std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&)
   void add_on_state_callback(std::function<void(std::string, size_t)> &&callback);
        ^
src/esphome/components/select/select.h:55:8: note:   no known conversion for argument 1 from 'esphome::panasonic_ac::PanasonicAC::set_vertical_swing_select(esphome::select::Select*)::<lambda(const string&)>' to 'std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&'
src/esphome/components/panasonic_ac/esppac.cpp: In member function 'void esphome::panasonic_ac::PanasonicAC::set_horizontal_swing_select(esphome::select::Select*)':
src/esphome/components/panasonic_ac/esppac.cpp:196:4: error: no matching function for call to 'esphome::select::Select::add_on_state_callback(esphome::panasonic_ac::PanasonicAC::set_horizontal_swing_select(esphome::select::Select*)::<lambda(const string&)>)'
   });
    ^
In file included from src/esphome/components/panasonic_ac/esppac.h:4:0,
                 from src/esphome/components/panasonic_ac/esppac.cpp:1:
src/esphome/components/select/select.h:55:8: note: candidate: void esphome::select::Select::add_on_state_callback(std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&)
   void add_on_state_callback(std::function<void(std::string, size_t)> &&callback);
        ^
src/esphome/components/select/select.h:55:8: note:   no known conversion for argument 1 from 'esphome::panasonic_ac::PanasonicAC::set_horizontal_swing_select(esphome::select::Select*)::<lambda(const string&)>' to 'std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&'
*** [.pioenvs/living-room-ac/src/esphome/components/panasonic_ac/esppac.cpp.o] Error 1
====================================================================== [FAILED] Took 8.15 seconds ======================================================================
hp@hpix:~$ 
hp@hpix:~$ esphome clean livingroom-ac.yaml
INFO Reading configuration livingroom-ac.yaml...
INFO Detected timezone 'Etc/UTC'
INFO Deleting .esphome/build/living-room-ac/.pioenvs
INFO Done!
hp@hpix:~$ esphome compile livingroom-ac.yaml
INFO Reading configuration livingroom-ac.yaml...
INFO Detected timezone 'Etc/UTC'
INFO Generating C++ source...
INFO Compiling app...
Processing living-room-ac (board: esp32cam; framework: arduino; platform: platformio/espressif32 @ 3.5.0)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
Dependency Graph
|-- <WiFi> 1.0
|-- <ESPmDNS> 1.0
|-- <Update> 1.0
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_connection.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_frame_helper.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_pb2.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_pb2_service.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/api_server.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/list_entities.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/proto.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/subscribe_state.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/api/user_services.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/binary_sensor/automation.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/binary_sensor/binary_sensor.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/binary_sensor/filter.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/climate/climate.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/climate/climate_mode.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/climate/climate_traits.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/esp32/core.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/esp32/gpio_arduino.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/esp32/gpio_idf.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/esp32/preferences.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/integration/integration_sensor.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/logger/logger.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/md5/md5.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_component.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_esp32_arduino.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_esp8266.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/mdns/mdns_esp_idf.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/network/util.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_backend_arduino_esp32.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_backend_arduino_esp8266.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_backend_esp_idf.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/ota/ota_component.cpp.o
Compiling .pioenvs/living-room-ac/src/esphome/components/panasonic_ac/esppac.cpp.o
src/esphome/components/panasonic_ac/esppac.cpp: In member function 'void esphome::panasonic_ac::PanasonicAC::set_vertical_swing_select(esphome::select::Select*)':
src/esphome/components/panasonic_ac/esppac.cpp:187:4: error: no matching function for call to 'esphome::select::Select::add_on_state_callback(esphome::panasonic_ac::PanasonicAC::set_vertical_swing_select(esphome::select::Select*)::<lambda(const string&)>)'
   });
    ^
In file included from src/esphome/components/panasonic_ac/esppac.h:4:0,
                 from src/esphome/components/panasonic_ac/esppac.cpp:1:
src/esphome/components/select/select.h:55:8: note: candidate: void esphome::select::Select::add_on_state_callback(std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&)
   void add_on_state_callback(std::function<void(std::string, size_t)> &&callback);
        ^
src/esphome/components/select/select.h:55:8: note:   no known conversion for argument 1 from 'esphome::panasonic_ac::PanasonicAC::set_vertical_swing_select(esphome::select::Select*)::<lambda(const string&)>' to 'std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&'
src/esphome/components/panasonic_ac/esppac.cpp: In member function 'void esphome::panasonic_ac::PanasonicAC::set_horizontal_swing_select(esphome::select::Select*)':
src/esphome/components/panasonic_ac/esppac.cpp:196:4: error: no matching function for call to 'esphome::select::Select::add_on_state_callback(esphome::panasonic_ac::PanasonicAC::set_horizontal_swing_select(esphome::select::Select*)::<lambda(const string&)>)'
   });
    ^
In file included from src/esphome/components/panasonic_ac/esppac.h:4:0,
                 from src/esphome/components/panasonic_ac/esppac.cpp:1:
src/esphome/components/select/select.h:55:8: note: candidate: void esphome::select::Select::add_on_state_callback(std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&)
   void add_on_state_callback(std::function<void(std::string, size_t)> &&callback);
        ^
src/esphome/components/select/select.h:55:8: note:   no known conversion for argument 1 from 'esphome::panasonic_ac::PanasonicAC::set_horizontal_swing_select(esphome::select::Select*)::<lambda(const string&)>' to 'std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&'
*** [.pioenvs/living-room-ac/src/esphome/components/panasonic_ac/esppac.cpp.o] Error 1
====================================================================== [FAILED] Took 8.71 seconds ======================================================================
hp@hpix:~$
```

@DomiStyle @hpirila I have the same issue after upgrading ESPHome to 2022.5.0

```
INFO Reading configuration /config/esphome/bedroom-ac.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing bedroom-ac (board: esp01_1m; framework: arduino; platform: platformio/espressif8266 @ 3.2.0)
--------------------------------------------------------------------------------
HARDWARE: ESP8266 80MHz, 80KB RAM, 1MB Flash
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
Dependency Graph
|-- <ESP8266WiFi> 1.0
|-- <ESP8266mDNS> 1.2
Compiling /data/bedroom-ac/.pioenvs/bedroom-ac/src/esphome/components/panasonic_ac/esppac.cpp.o
Compiling /data/bedroom-ac/.pioenvs/bedroom-ac/src/esphome/components/select/select.cpp.o
Compiling /data/bedroom-ac/.pioenvs/bedroom-ac/src/esphome/components/select/select_call.cpp.o
Compiling /data/bedroom-ac/.pioenvs/bedroom-ac/src/esphome/components/select/select_traits.cpp.o
Compiling /data/bedroom-ac/.pioenvs/bedroom-ac/src/esphome/components/sensor/automation.cpp.o
src/esphome/components/panasonic_ac/esppac.cpp: In member function 'void esphome::panasonic_ac::PanasonicAC::set_vertical_swing_select(esphome::select::Select*)':
src/esphome/components/panasonic_ac/esppac.cpp:187:4: error: cannot convert 'esphome::panasonic_ac::PanasonicAC::set_vertical_swing_select(esphome::select::Select*)::<lambda(const string&)>' to 'std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&'
  187 |   });
      |    ^
In file included from src/esphome/components/panasonic_ac/esppac.h:4,
                 from src/esphome/components/panasonic_ac/esppac.cpp:1:
src/esphome/components/select/select.h:55:73: note:   initializing argument 1 of 'void esphome::select::Select::add_on_state_callback(std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&)'
   55 |   void add_on_state_callback(std::function<void(std::string, size_t)> &&callback);
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
src/esphome/components/panasonic_ac/esppac.cpp: In member function 'void esphome::panasonic_ac::PanasonicAC::set_horizontal_swing_select(esphome::select::Select*)':
src/esphome/components/panasonic_ac/esppac.cpp:196:4: error: cannot convert 'esphome::panasonic_ac::PanasonicAC::set_horizontal_swing_select(esphome::select::Select*)::<lambda(const string&)>' to 'std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&'
  196 |   });
      |    ^
In file included from src/esphome/components/panasonic_ac/esppac.h:4,
                 from src/esphome/components/panasonic_ac/esppac.cpp:1:
src/esphome/components/select/select.h:55:73: note:   initializing argument 1 of 'void esphome::select::Select::add_on_state_callback(std::function<void(std::__cxx11::basic_string<char>, unsigned int)>&&)'
   55 |   void add_on_state_callback(std::function<void(std::string, size_t)> &&callback);
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
Compiling /data/bedroom-ac/.pioenvs/bedroom-ac/src/esphome/components/sensor/filter.cpp.o
Compiling /data/bedroom-ac/.pioenvs/bedroom-ac/src/esphome/components/sensor/sensor.cpp.o
*** [/data/bedroom-ac/.pioenvs/bedroom-ac/src/esphome/components/panasonic_ac/esppac.cpp.o] Error 1
========================= [FAILED] Took 11.06 seconds =========================

```

--- a/components/panasonic_ac/esppac.cpp
+++ b/components/panasonic_ac/esppac.cpp
@@ -180,7 +180,7 @@ void PanasonicAC::set_current_temperature_sensor(sensor::Sensor *current_tempera
 
 void PanasonicAC::set_vertical_swing_select(select::Select *vertical_swing_select) {
   this->vertical_swing_select_ = vertical_swing_select;
-  this->vertical_swing_select_->add_on_state_callback([this](const std::string &value) {
+  this->vertical_swing_select_->add_on_state_callback([this](const std::string &value, size_t index) {
     if (value == this->vertical_swing_state_)
       return;
     this->on_vertical_swing_change(value);
@@ -189,7 +189,7 @@ void PanasonicAC::set_vertical_swing_select(select::Select *vertical_swing_selec
 
 void PanasonicAC::set_horizontal_swing_select(select::Select *horizontal_swing_select) {
   this->horizontal_swing_select_ = horizontal_swing_select;
-  this->horizontal_swing_select_->add_on_state_callback([this](const std::string &value) {
+  this->horizontal_swing_select_->add_on_state_callback([this](const std::string &value, size_t index) {
     if (value == this->horizontal_swing_state_)
       return;
     this->on_horizontal_swing_change(value);

